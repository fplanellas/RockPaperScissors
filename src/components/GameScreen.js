import React, { createContext, useState } from 'react';
import Header from "./Header";
import styled from 'styled-components';
import Wrapper from './Wrapper';
import GameBoard from './GameBoard';

const ScreenStyled = styled.main`
    background: #000;
    padding: 0.5em;
    position: relative;
    height: 500px;
    color:#fff;
  `;

export const ScoreContex = createContext();
export const GameBoardContex = createContext();
export const GameBoardInit = createContext();

const GameScreen = () => {

    const [score, setScore] = useState(0);
    const [gameB, setGameB] = useState(true);
    const [userPlaying, setUserPlaying] = useState(false);

    return (
        <GameBoardInit.Provider value={{
            userPlaying,
            setUserPlaying
        }}>
            <GameBoardContex.Provider value={{
                gameB,
                setGameB
            }}>
                <ScoreContex.Provider value={{
                    score,
                    setScore
                }}>
                    <ScreenStyled>
                        <Wrapper>
                            <Header />
                            <GameBoard />
                        </Wrapper>
                    </ScreenStyled>
                </ScoreContex.Provider>
            </GameBoardContex.Provider>
        </GameBoardInit.Provider>
    );
};

export default GameScreen;
