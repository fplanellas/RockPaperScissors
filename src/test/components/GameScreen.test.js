import React from 'react';
import { shallow } from 'enzyme';
import GameScreen from '../../components/GameScreen';

describe('testing in component GameScreen', () => {

    test('should be displayed correctly', () => {
        const wrapper = shallow( <GameScreen /> );
        expect( wrapper ).toMatchSnapshot();
    })
      
})