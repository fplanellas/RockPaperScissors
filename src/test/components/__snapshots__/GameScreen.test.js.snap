// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing in component GameScreen should be displayed correctly 1`] = `
<ContextProvider
  value={
    Object {
      "setUserPlaying": [Function],
      "userPlaying": false,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "gameB": true,
        "setGameB": [Function],
      }
    }
  >
    <ContextProvider
      value={
        Object {
          "score": 0,
          "setScore": [Function],
        }
      }
    >
      <styled.main>
        <Wrapper>
          <Header />
          <GameBoard />
        </Wrapper>
      </styled.main>
    </ContextProvider>
  </ContextProvider>
</ContextProvider>
`;
